packages/wallet/src/KeyManagement/types.ts

```diff
@@ -134,7 +138,10 @@ export interface KeyAgent {
   /**
    * @throws AuthenticationError
    */
-  signTransaction(txInternals: TxInternals, options: SignTransactionOptions): Promise<Cardano.Signatures>;
+  signTransaction(
+    txInternals: TxInternals,
+    options: SignTransactionOptions | SignHardwareTransactionOptions
```

Comment:
Maybe just as a guard, and to then map to the `NetworkId`? https://github.com/cardano-foundation/ledgerjs-hw-app-cardano/blob/2b049051ffd4193ac6c713e9bfd19521ddf45525/src/interactions/serialization/txInit.ts#L47

Reply:
> I'm leaning towards adding it to SerializableTrezorKeyAgentData and SerializableLedgerKeyAgentData instead of base.

Agree. It would be a misdirection if it's not needed for all key agents.

Reply:
@rhyslbw @mkazlauskas 
Idea was to prevent passing networkInforProvider and its data to key agents if is not necessary. Since networkInforProvider lives in the scope of SingleAddressWallet idea was to just pass it to the key agent where and when is necessary. Do we really need to add it to the serializable data and always instantiate hw key agents with the provider already passed to the wallet?

Reply:
I agree it is more convenient in context of single address wallet, but we have to consider that key agent can also be used outside of SingleAddressWallet, in which case having to pass around those static values might not be as convenient.

Both network ID and network magic could be just hardcoded by the app that uses the key agents, or they could choose to fetch it using NetworkInfoProvider.

Reply:
@mkazlauskas 👌 
Thanks for the clarification